src/pages/Intents/CustomIntents.js

```diff
@@ -0,0 +1,80 @@
+import { useDataMutation } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import isEmpty from 'lodash/isEmpty'
+import isEqual from 'lodash/isEqual'
+import React, { useEffect, useMemo, useState } from 'react'
+import { useIsAuthorized } from '../../auth'
+import FooterStripButtons from '../../components/footerStripButton/FooterStripButtons'
+import { CustomIntentInfo } from '../../components/noticeAlert'
+import Page from '../../components/page/Page'
+import CustomIntentsList from './CustomIntentsList'
+import {
+    saveCustomIntentsMutation,
+    useReadCustomIntentsDataStore,
+} from './intentsDatastoreQuery'
+import { IntentsList } from './mockList'
+
+//TODO: change IntentsList mock data to real data
+const CustomIntents = () => {
+    const { hasAuthority } = useIsAuthorized()
+    const { customIntents } = useReadCustomIntentsDataStore()
+    const [disableSave, setDisableSave] = useState(true)
+    const [customIntentSettings, setCustomIntentSettings] =
+        useState(IntentsList)
+    const initialValues = useMemo(() => {
+        return customIntents
+    }, [customIntentSettings])
+    const [mutate, { error, data }] = useDataMutation(saveCustomIntentsMutation)
```

Comment:
It might also make sense to just export the mutation from the a common hook (e.g. combine this and `useReadCustomIntentsDataStore` into something more like `useCustomIntents`)